src/ai/defaultai.cc

```diff
@@ -4887,14 +4861,14 @@ bool DefaultAI::check_mines_(uint32_t const gametime) {
 	bool single_critical = false;
 	if ((site.bo->is(BuildingAttribute::kBuildingMatProducer) ||
 	     site.bo->mines == iron_resource_id) &&
-	    mines_per_type[site.bo->mines].total_count() == 1) {
+	    mines_per_type[site.bo->mines].finished == 1) {
```

Comment:
the logic changes here, total vs finished - was it intentional?

Reply:
yes before a single critical mine could be dismanteled when another was in construction this led to constant rebuild and could lead to a stuck situation if not enough wares for the construction of the mine